lib/screens/explore_screens/regions_screens/join_region/components/not_enough_seeds_dialog.dart

```diff
@@ -9,6 +9,10 @@ import 'package:url_launcher/url_launcher.dart';
 class NotEnoughSeedsDialog extends StatelessWidget {
   const NotEnoughSeedsDialog({super.key});
 
+  Future<void> show(BuildContext context) async {
```

Comment:
i think this should be moved to the base dialog

instead of adding this show method to all dialogs

Reply:
import 'dart:io';

import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:mindset/v2/core/extensions/scope_functions.dart';
import 'package:mindset/v2/design/messages/message_type.dart';

class MindsetDialog {
  final String title;
  final String? contentText;
  final String primaryActionText;
  final VoidCallback primaryActionCallback;
  final String? secondaryActionText;
  final VoidCallback? secondaryActionCallback;
  final MessageType messageType;

  const MindsetDialog({
    required this.title,
    this.contentText,
    required this.primaryActionText,
    required this.primaryActionCallback,
    this.secondaryActionText,
    this.secondaryActionCallback,
    this.messageType = MessageType.success,
  });

  void show(BuildContext context) {
    if (Platform.isIOS) {
      showIOSDialog(context);
    } else {
      showAndroidDialog(context);
    }
  }

  void showAndroidDialog(BuildContext context) {
    final actions = [
      Padding(
        padding: const EdgeInsets.all(8),
        child: TextButton(
          onPressed: primaryActionCallback,
          child: Text(primaryActionText),
        ),
      )
    ];
    secondaryActionText?.let(
      (self) => actions.add(Padding(
        padding: const EdgeInsets.all(8),
        child: TextButton(
          onPressed: secondaryActionCallback,
          child: Text(self),
        ),
      )),
    );

    AlertDialog(
      title: Text(title),
      content: contentText?.let((self) => Text(self)),
      actions: actions,
    ).let(
      (self) => {showDialog(context: context, builder: (context) => self)},
    );
  }

  void showIOSDialog(BuildContext context) {
    final actions = [
      CupertinoDialogAction(
        onPressed: primaryActionCallback,
        child: Text(primaryActionText),
      )
    ];
    secondaryActionText?.let(
      (self) => actions.add(CupertinoDialogAction(
        onPressed: secondaryActionCallback,
        child: Text(self),
      )),
    );

    CupertinoAlertDialog(
      title: Text(title),
      content: contentText?.let((self) => Text(self)),
      actions: actions,
    ).let(
      (self) => {showDialog(context: context, builder: (context) => self)},
    );
  }
}

Reply:
look into MindsetDialog in the mindset repo

Reply:
That's something I wanted to do a long time ago, but I wasn't sure which OOP tool I was arivia ahi haha si extends or implements. great I will give you a revision although it would be a big change I do not think it should be done in this PR

Reply:
> > > Nice Raul 👍 , Just one small thing I notice the Dialogs are not center Can you fix this as part of this pr? ![image](https://user-images.githubusercontent.com/51983923/176939088-d380c944-45a7-441e-bc67-c9ee2d655d05.png)
> > 
> > 
> > Sorry but not sure what part do you mean needs be centered I see the dialog in the middle of the screen.
> 
> The full dialog. Is not center with the rest of the screen.

Could you fix it? I don't really see the bad centering in fact I see a center widget in the base dialog.